src/Web/Event/CustomEvent.purs

```diff
@@ -12,3 +13,15 @@ fromEvent = unsafeReadProtoTagged "CustomEvent"
 
 toEvent :: CustomEvent -> Event
 toEvent = U.unsafeCoerce
+
+foreign import new :: String -> CustomEvent
+
+-- | Create a new `CustomEvent`, storing some data in its `detail` field,
+-- | and using defaults for everything else.
+new' :: forall a. String -> Nullable a -> CustomEvent
+new' ty det = newOptions ty { detail: det, bubbles: false, cancelable: false, composed: false }
+
+-- | Create a new `CustomEvent` with all of its constructor's options exposed.
+foreign import newOptions :: forall a. String -> { detail :: Nullable a , bubbles :: Boolean, cancelable :: Boolean, composed :: Boolean } -> CustomEvent
```

Comment:
If we're going to do both an options variant and a non-options variant, then perhaps we should stick to the same naming scheme we've been using recently, ie. `newWithOptions`.